.github/actions/terraform-observe/opal-validation/action.yaml

```diff
@@ -0,0 +1,87 @@
+name: 'OPAL validation'
+description: 'run tf apply and destroy to validate opal'
+inputs:
+  observe-customer-id:
+    description: "customer id for test observe instance"
+    required: true
+  observe-domain:
+    description: "observeinc.com or observe-staging.com"
+    required: true
+  observe-user-handle: 
+    description: "observe user to create test assets"
+    required: true
+  observe-user-pswd:
+    description: "password for user to create test assets"
+    required: true
+    
+runs:
+  using: "composite"
+  steps:
+
+    - id: install-terraform
+      uses: hashicorp/setup-terraform@v1
+
+    - id: get-github-repo-name
+      run: echo "REPOSITORY_NAME=$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')" >> $GITHUB_ENV
+      shell: bash
+
+    - id: checkout-commit
+      uses: actions/checkout@v2
+
+    # for every directory under /tests/ ...
+    # cd to directory
+    # terraform init, plan, apply, (eventually other things), destroy
+
+    # we could run all the terraform commands (init, plan, apply, destroy) in one single step.
+    # this would save us the theoretical trouble of having all the test modules applied at once 
+    # and then destroyed later. But if we do this we woudl have to run any extra assertions within
+    # that single step between apply + destroy. It seems more clearly valuable that we would keep 
+    # the custom assertions in their own separate step, since that wuold make troubleshooting test
+    # failures easier
+    
+    - id: tf-setup
+      shell: bash
+      run: |
+        ran_init=false
+        for dir in .tests/*/ ; do
+          cd ${{ github.workspace }}/$dir
+          # save everyone the need to create boilerplate provider.tf files in their tests
+          # uses cat since passwords can contain chars that makes sed sad (&\)
+          cat <<EOF > ci_added_provider.tf
+          provider "observe" {
+          customer      = "${{ inputs.observe-customer-id }}"
```

Comment:
You don't need this. Just ensure that provider is not set and configure the environment variable?

Reply:
wow, didn't realize we could set these with env vars. That's nice!